grype/pkg/metadata.go

```diff
@@ -0,0 +1,12 @@
+package pkg
+
+// MetadataType represents the data shape stored within pkg.Package.Metadata.
+type MetadataType string
+
+const (
+	// this is the full set of data shapes that can be represented within the pkg.Package.Metadata field
+
+	UnknownMetadataType MetadataType = "UnknownMetadata"
```

Comment:
Should this be just serialized during output based on the actual metadata data type?

Reply:
That's how this originally worked, yes. I decided to make a more firm division of responsibilities here between syft and grype since the data shapes represented by these constants are different depending which application being referred to.